client-scripts-api.md

```diff
@@ -277,3 +277,111 @@ Payload:
   value: true|false
 }
 ```
+
+### `routeChange`
+
+This event is triggered when a change in the internal router happens.
+
+Payload:
+
+```js
+{
+  location: {
+    pathname: '/products',
+    search: '?page=0',
+    query: {
+      page: 0,
+    },
+    hash: '#',
+  },
+  action: 'PUSH',
+}
+```
+
+### `searchTermChange`
+
+Subject changed and applied a new search term. Includes also the result count for the new search.
+
+Payload:
+
+```js
+{
+  searchTerm: 'chocolate',
+  resultCount: 145,
+}
+```
+
+The property `resultCount` is optional and could be omitted if result count is not known or not available.
+
+### `resultsFilterChange`
+
+Subject changed the filters for results and applied the filters or initial empty filter has been applied.
+
+Payload:
+
+```js
+{
+  activeFilters: {
+    brandName: ['snickers', 'milka', 'lindt', 'm&ms'],
+  }
+  activeFilterCount: 4,
+  resultCount: 46, // Results after filter has been applied
+}
+```
+
+### `paginationChange`
+
+Results page has been changed or initialized. Origin of event tells where the pagination has changed.
+
+Payload:
+
+```js
+{
+  hasNextPage: true
+  hasPrevPage: true
+  page: 6
+  totalPageCount: 59
+}
+```
+
+### E-Commerce specific events
+
```

Comment:
❓ what about events like `addToBasked`?

Reply:
They are all done with the `basketQuantityChange` event to simplify it. From the `difference` property you can see if it was newly added.